test/suites/basic.sh

```diff
@@ -618,4 +618,27 @@ test_basic_usage() {
   lxc storage volume delete bla vol1
   lxc storage volume delete bla vol2
   lxc storage delete bla
+
+  # Test rebuilding an instance with its original image.
+  lxc init testimage c1
+  lxc start c1
+  lxc exec c1 -- touch /data.txt
+  lxc stop c1
+  lxc rebuild c1
+  lxc start c1
+  ! lxc exec c1 -- ls -lh /data.txt || false
+  lxc delete c1 -f
+
+  # Test rebuilding an instance with a new image.
+  lxc init c1 --empty
+  lxc_remote remote add l1 "${LXD_ADDR}" --accept-certificate --password foo
+  lxc_remote rebuild l1:testimage c1
+  lxc start c1
+  lxc delete c1 -f
+  lxc_remote remote remove l1
+
+  # Test rebuilding an instance with an empty file system.
+  lxc init testimage c1
```

Comment:
You should be able to use `lxc file push` to push a file into a stopped empty container, so you can check its gone after rebuild using `lxc file pull`.

Reply:
@tomponline actually I get a `Failed to statfs` because as the container is empty I don't have a rootfs..